patroni.py

```diff
@@ -88,6 +91,11 @@ def run(self):
             try:
                 if self.ha.state_handler.is_leader():
                     self.ha.cluster and self.ha.state_handler.create_replication_slots(self.ha.cluster)
+
+                    # SkyDNS2 support: publish leader
```

Comment:
That was my thinking as well but for our use case callback cannot be used because it's being called just once - on a role change. We need periodic record refresh to keep record's TTL short = faster response when anything goes wrong.

Sorry for that, it wasn't meant to be part of my PR. Still being a git noob, I don't know how to divide future commits from PR I've already made.. :-(